trackers/core/deepsort/feature_extractor.py

```diff
@@ -105,14 +105,43 @@ def from_timm(
         backbone_model = FeatureExtractionBackbone(model)
         return cls(backbone_model, device, input_size)
 
+    def _download_checkpoint_from_url(self, url: str) -> str:
+        """Downloads a model checkpoint from a URL.
+
+        Args:
+            url (str): The URL to download the model checkpoint from.
+
+        Returns:
+            str: The local path to the downloaded checkpoint file.
+        """
+        from trackers.utils.download_file_utils import AsyncFileDownloader
+
+        downloader = AsyncFileDownloader()
+        try:
+            loop = asyncio.get_event_loop()
+            if loop.is_running():
+                future = asyncio.ensure_future(downloader.process_url(url))
+                checkpoint_path = loop.run_until_complete(future)
+            else:
+                checkpoint_path = loop.run_until_complete(downloader.process_url(url))
+        except RuntimeError:
+            checkpoint_path = asyncio.run(downloader.process_url(url))
+        print(
+            f"Model downloaded to {checkpoint_path}. "
+            "Loading the model from the checkpoint."
+        )
+        return checkpoint_path
```

Comment:
Hi @onuralpszr, can you rename this function to `download_from_url` and move it to `trackers/utils/download_file_utils.py`? I think it would be better if this function is available as a general utility.

Reply:
It is done.